[sdk 33] SDK 33 candidate

ide opened the pull request:
# Why

This is the candidate branch for SDK with Android and iOS updated. It uses RN 0.59.6 and React 16.8.6. There is still work to do and SDK 33 is not complete but this will let us bring master up to speed.

# Test Plan

- Loaded Home on Android and iOS
- Ran Jest tests in the expo package -> all passed
- Verifying that CI passes

ide committed:
[sdk33] Update "react-native" in package.json to 0.59, "react" to 16.8.3

Make each app use RN 0.59.

tsapeta committed:
[gl-cpp] configure expo-gl-cpp for 64 bit architecture and new JSC

tsapeta committed:
[gl] make expo-gl-cpp a dependency of expo-gl

ide committed:
[sdk33] Update iOS with RN 0.59

ide committed:
[sdk33] Temporarily set assetPlugins within this repo

tsapeta committed:
revert changes in UEXGL code

tsapeta committed:
replace gnustl_shared with c++_shared

ide committed:
[sdk33] Update ReactAndroid and ReactCommon

ide committed:
[sdk33] Update Android Gradle plugin, Kotlin, and Butter Knife

The new ReactAndroid Gradle file uses a newer `variant.javaCompileProvider` API, so I updated the Android Gradle plugin. In turn, I needed to upgrade Kotlin and Butter Knife to newer versions that support the newer Android Gradle plugin.

ide committed:
[sdk33] Update React Navigation, remove unused Detox resolution, remove unused side effects

- Updated React Navigation to the latest in Home
- Removed Detox from "resolutions" in the workspace root, since we now just have one copy of Detox. This is because we remove devDependencies from RN in our fork.
- Removed unused side effects files under `expo/src/environment` (these files were renamed but the old copies weren't removed)

Tested by loading Home on iOS.

wkozyra95 committed:
[gl] extract JSContextRef from JSCRuntime

wkozyra95 committed:
[gl] revert non backward compatibile changes

tsapeta committed:
Update packages/expo-gl-cpp/android/src/main/jni/EXGL.cpp

Co-Authored-By: wkozyra95 <wkozyra95@gmail.com>

ide committed:
[sdk33][jest] Update jest-expo's jest-preset to handle RN's new JS preset

RN moved from JSON to JS for its Jest preset, so we need to change ours now. We previously generated our Jest preset ahead of time based on RN's JSON, but now our preset is JS as well and reads in and transforms RN's preset at runtime.

Tested by running the Jest tests in the expo package.

The path to the assetFileTransformer wasn't right, causing Home tests to fail. Now Home tests pass.

Fail fast and often

ide committed:
[sdk33] Update to Jest 24 and a compatible version of ts-jest

- Upgraded to the new version of Jest that RN uses
- Upgraded to a compatible version of ts-jest
- Memoized UIManager in jest-expo's setup.js so that calling jest.resetModules() doesn't break things. The correct fix later is to make RN's UIManager.js idempotent.

Test plan: Ran yarn test in the expo package.

tsapeta committed:
[gl] make expo-gl-cpp compatible with all SDKs

tsapeta committed:
[gl] update expo-gl-cpp binaries and add 64 bit ones

tsapeta committed:
[gl] remove `libexpo-gl.so` from versioned expo-gl aar archives

tsapeta committed:
[sdk33] update react-native in SDK31-32 to be linked with libc++_shared standard library

tsapeta committed:
[sdk33] fix android-build-aar.sh script to remove libc++_shared binaries from aar

ide committed:
[sdk33] Update react-native to the sdk-33 branch

ide committed:
[sdk33] Fix expo-asset unit test (assert that img.onerror is defined)

ide committed:
[sdk33][android] Update compileSdkVersion to 28 to match RN

ide committed:
[sdk33] Publish new dev home for SDK 33

tsapeta committed:
[sdk33] fix ReactAndroid

tsapeta committed:
[sdk33] make sure generateDynamicMacros is executed before all other tasks

ide committed:
[sdk33] Upgrade all SoLoader references to 0.6.0

SoLoader was upgraded to 0.6.0 in RN 0.59. It is backwards-compatible with SoLoader 0.5.1, which was not backwards-compatible with SoLoader 0.1.0.

This is OK, since RN 0.57 upgraded to SoLoader 0.5.1 and that is the oldest RN version we are including in the next Expo Client with SDK 31 & 32. We are dropping SDK 30 and older, which used RN 0.55.

So, we don't need to patch SoLoader 0.6.1 with hand-written JVM bytecode (see https://github.com/expo/expo/commit/7b5d52ed77fed9d6bf7c0d55ba0608b4341787ec for the patch) and include it in our local Maven repository and can just use the regular copy from Maven Central. Also, this means we can use a newer javac, which disallows .class files with two methods that differ only by return type -- the technique we were relying on to simultaneously support SoLoader 0.1.0 and 0.5.1 previously.

eveningkid commented:
Good luck guys and thanks for doing this. I am really excited about this update :)

ide committed:
[sdk33] Replace Android network config with `android:usesCleartextTraffic` for dev

Android 9.0 blocks all cleartext (HTTP) traffic by default, which breaks the connection with local development servers. We had a network security config before for FBAds, but it is too fine-grained and allows whitelisting only specific domains rather than ranges of IP addresses. I verified by reading the Android 9 source code that there is no way to specify an IP range: http://androidxref.com/9.0.0_r3/xref/frameworks/base/core/java/android/security/net/config/ApplicationConfig.java.

Tested by building the Expo client for an Android 9 simulator and successfully loading test-suite. Without this change, the client fails to connect tot the dev server and the error log says `Uncaught Error: java.net.UnknownServiceException: CLEARTEXT communication to <ip> not permitted by network security policy`.

netpoe commented:
Can we use this already? I mean, does `expo init` will use this version right away?

tsapeta commented:
No, this PR only updates react-native to v0.59. There is still some work ahead of us before we roll out SDK32. We will announce that soon.

I'm locking this conversation, if you have more questions please ask us on forums.
